src/main/java/uk/nhs/digital/nhsconnect/lab/results/configuration/CustomTrustStore.java

```diff
@@ -0,0 +1,104 @@
+package uk.nhs.digital.nhsconnect.lab.results.configuration;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3URI;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import lombok.NoArgsConstructor;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+import java.security.KeyStore;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+@Component
+@Slf4j
+@NoArgsConstructor
+public class CustomTrustStore {
+
+    @Autowired(required = false)
+    private AmazonS3 s3Client;
+
+    @SneakyThrows
+    public void addToDefault(String trustStorePath, String trustStorePassword) {
+        final X509TrustManager defaultTrustManager = getDefaultTrustManager();
+        final X509TrustManager customTrustManager =
+            getCustomDbTrustManager(new AmazonS3URI(trustStorePath), trustStorePassword);
+        X509TrustManager combinedTrustManager = new CombinedTrustManager(customTrustManager, defaultTrustManager);
+
+        SSLContext sslContext = SSLContext.getInstance("TLS");
+        sslContext.init(null, new TrustManager[] {combinedTrustManager}, null);
+        LOGGER.info("Overriding default TrustStore with combined one");
+        SSLContext.setDefault(sslContext);
+    }
+
+    @SneakyThrows
+    private X509TrustManager getDefaultTrustManager() {
+        TrustManagerFactory trustManagerFactory =
+            TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        trustManagerFactory.init((KeyStore) null); // Using null here initialises the TMF with the default trust store.
+
+        for (TrustManager tm : trustManagerFactory.getTrustManagers()) {
+            if (tm instanceof X509TrustManager) {
+                return (X509TrustManager) tm;
+            }
+        }
+        throw new IllegalStateException("Cannot find trust manager");
```

Comment:
May be more idiomatic Java 8+.

```suggestion
        return trustManagerFactory.getTrustManagers().stream()
            .filter(X590TrustManager.class::isInstance)
            .map(X590TrustManager.class::cast)
            .findAny()
            .orElseThrow(() -> new IllegalStateException("Cannot find trust manager"));
```

Also this block is repeated in the next method - candidate for an extract method refactor